healthcheck/healthcheck_test.go

```diff
@@ -193,6 +193,64 @@ func TestSetBackendsConfiguration(t *testing.T) {
 	}
 }
 
+func TestNewRequest(t *testing.T) {
+	tests := []struct {
+		host     string
+		port     int
+		path     string
+		expected string
+	}{
+		{
+			host:     "backend1:80",
+			port:     0,
+			path:     "/test",
+			expected: "http://backend1:80/test",
+		},
+		{
+			host:     "backend2:80",
+			port:     8080,
+			path:     "/test",
+			expected: "http://backend2:8080/test",
+		},
+		{
+			host:     "backend1",
+			port:     0,
+			path:     "/health",
+			expected: "http://backend1/health",
+		},
+		{
+			host:     "backend2",
+			port:     8080,
+			path:     "/health",
+			expected: "http://backend2:8080/health",
+		},
+	}
+
+	for _, test := range tests {
+		backend := NewBackendHealthCheck(
```

Comment:
Sure. I wasn't sure it was worth the effort, but I can make it match the others.